src/libstore/package.nix

```diff
@@ -21,6 +21,10 @@
   version,
 
   embeddedSandboxShell ? stdenv.hostPlatform.isStatic,
+
+  withAWS ?
+    # Default is this way because there have been issues building this dependency
+    stdenv.hostPlatform == stdenv.buildPlatform && (stdenv.isLinux || stdenv.isDarwin),
```

Comment:
This didn't apply to the everything package or package set.
Instead, Nixpkgs upstream can now do
```nix
nix.overrideScope (finalScope: prevScope: { aws-sdk-cpp = null; })
```
or
```nix
nixComponents.overrideScope (finalScope: prevScope: { aws-sdk-cpp = null; })
```

Should we keep `withAWS`? It would only work for `nix-store`, via `overrideAllMesonComponents`, which is not great UX at this point.